src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -109,6 +109,28 @@ impl ScalarFunc {
         }
         Ok(Some(Cow::Owned(output)))
     }
+    #[inline]
+    pub fn concat_ws<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let mut output: Vec<u8> = Vec::new();
+        let mut output_sep: Vec<u8> = Vec::new();
+        let mut signal = true;
+        for expr in &self.children {
+            if signal {
+                let input = try_opt!(expr.eval_string(ctx, row));
+                output_sep.extend_from_slice(&input);
+                signal = false;
+            } else {
+                let input1 = try_opt!(expr.eval_string(ctx, row));
+                output.extend_from_slice(&input1);
+            }
+        }
+        output.join(out_sep);
```

Comment:
```
error[E0425]: cannot find value `out_sep` in this scope
   --> src/coprocessor/dag/expr/builtin_string.rs:131:21
    |
131 |         output.join(out_sep);
    |                     ^^^^^^^ not found in this scope

error[E0599]: no method named `join` found for type `std::vec::Vec<u8>` in the current scope
   --> src/coprocessor/dag/expr/builtin_string.rs:131:16
    |
131 |         output.join(out_sep);
    |                ^^^^
```

Do you mean `output_seq` here?

Reply:
https://stackoverflow.com/a/28312049/7083401

Reply:
@kg88